tlslite/extensions.py

```diff
@@ -1804,7 +1808,7 @@ def parse(self, parser):
         """Deserialise the extension from on the wire data."""
         # the generic class allows for missing values, it's not allowed here
         if not parser.getRemainingLength():
-            raise SyntaxError()
+            raise DecodeError("Non empty extension payload")
```

Comment:
Should not this say "Empty extension payload"? Or do I read the condition wrongly?

Reply:
no, here is exactly the opposite: the extension must be empty, so if it isn't, it's an error

Reply:
`not parser.getRemainingLength()` for `parser.getRemainingLength() == 1` is `false`, therefore it does not get into this condition. If `parser.getRemainingLength() == 0`, then the condition is `true` and we go into this condition -- extension is empty with zero length.
It is also consistent with what I read in the RFC -- the extension can not be empty as it needs to carry the modes.
Can you double check?

Reply:
Or the modes of this extension are parsed before calling this function and this checks for trailing data?

Reply:
facepalm.jpg, yes, you're right, it should be "Empty extension paylod"